filesystem/ext2.ksy

```diff
@@ -23,14 +23,15 @@ types:
       - id: super_block
         type: super_block_struct
         size: 1024
-      # http://www.nongnu.org/ext2-doc/ext2.html#BLOCK-GROUP-DESCRIPTOR-TABLE
+        doc-ref: http://www.nongnu.org/ext2-doc/ext2.html#BLOCK-GROUP-DESCRIPTOR-TABLE
       - id: block_groups
         type: bgd
         repeat: expr
         repeat-expr: super_block.block_group_count
```

Comment:
The URL http://www.nongnu.org/ext2-doc/ext2.html#BLOCK-GROUP-DESCRIPTOR-TABLE is **not** associated with the field `super_block` but rather with the following field `block_groups`.

Please use your brain a bit - if the URL screams <code>http\://www.nongnu.org/ext2-doc/ext2.html#**BLOCK-GROUP**-DESCRIPTOR-TABLE</code>, don't bind it to the previous attribute.

Reply:
I just redid it in-place, so blame the original author.

Reply:
> I just redid it in-place, so blame the original author.

Not really. The _original author_ was just consistently putting reference links always **above** the concerned element. This makes quite sense, because it follows the convention common in programming languages, for example see https://google.github.io/styleguide/jsguide.html#jsdoc-method-and-function-comments:

```js
/** A class that does something. */
class SomeClass extends SomeBaseClass {
  /**
   * Operates on an instance of MyClass and returns something.
   * @param {!MyClass} obj An object that for some reason needs detailed
   *     explanation that spans multiple lines.
   * @param {!OtherClass} obviousOtherClass
   * @return {boolean} Whether something occurred.
   */
  someMethod(obj, obviousOtherClass) { ... }

  /** @override */
  overriddenMethod(param) { ... }
}

/**
 * Demonstrates how top-level functions follow the same rules.  This one
 * makes an array.
 * @param {TYPE} arg
 * @return {!Array<TYPE>}
 * @template TYPE
 */
function makeArray(arg) { ... }
```

Notice that the class and function comments also always come **above** the class or function. If we weren't using YAML and didn't have designated keys `doc` and `doc-ref` for these comments, I'd also recommend this approach. And I'd still use it on places where `doc` and `doc-ref` can't be used, i.e. for enum declarations:

```ksy
enums:
  # https://example.com/fruit-enum.html
  fruit:
    1: apple
    2: banana
```

And you can't just thoughtlessly do some find-and-replace `# http` with `doc-ref: http` and expect to get reasonable results. You always need to check what you're committing, especially when you used some automated tools to make them. _Find and replace_ is only a tool to facilitate tedious work, it won't autonomously work in place of you. It's like buying a Tesla Autopilot and climbing into the back seat while driving, leaving the driver seat empty and letting the vehicle to drive on its own - that's something incredibly stupid, because that not only endangers your own life but also the lives of other people.